### PR TITLE
Fixes - #1195

### LINKED ISSUE #1195: [Vi Mode] Shift+W at start of prompt in "command mode" always causes an exception
### Environment
PSReadLine: 2.0.0-beta5
PowerShell: 7.0.0-preview.5
OS: Microsoft Windows 10.0.19023
Last 87 Keys
```
 LeftWindows LeftWindows LeftWindows g c i Spacebar - p a t h Spacebar : Shift+Backspace C : \ p r o Tab Enter RightArrow Spacebar - r e Tab Spacebar | Shift+Spacebar G e t - F i l Tab Enter Enter LeftWindows G e t - E r r o r Enter UpArrow UpArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow RightArrow Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Enter UpArrow UpArrow Enter Escape W
```

### Exception
```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at System.Text.StringBuilder.get_Chars(Int32 index)
   at Microsoft.PowerShell.PSConsoleReadLine.ConvertOffsetToPoint(Int32 offset)
   at Microsoft.PowerShell.PSConsoleReadLine.MoveCursor(Int32 newCursor)
   at Microsoft.PowerShell.PSConsoleReadLine.ViNextGlob(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(PSKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics, CancellationToken)

```

### Steps to reproduce:

```powershell
Set-PSReadLineOption -EditMode Vi
```

<kbd>Enter</kbd>

<kbd>Esc</kbd>

<kbd>Shift</kbd>+<kbd>W</kbd>

### PR DESCRIPTION
**Pull request by springcomp**
Fixes a condition where the cursor could be set to [an incorrect position](https://github.com/PowerShell/PSReadLine/issues/1195#issuecomment-555864290).

Fix #1195

### CODE REVIEW

**Comment by springcomp** [comment]
This has been closed by mistake, I reckon.

File: PSReadLine/Movement.vi.cs [hunk 1]
```diff
@@ -94,7 +94,15 @@ public static void ViNextGlob(ConsoleKeyInfo? key = null, object arg = null)
                 i = _singleton.ViFindNextGlob(i);
             }
 
-            _singleton.MoveCursor(Math.Min(i, _singleton._buffer.Length - 1));
+            var newPosition = Math.Min(i, Math.Max(0, _singleton._buffer.Length - 1));
```
Comment (rjmholt) [inline PSReadLine/Movement.vi.cs]:
```suggestion
            int newPosition = Math.Min(i, Math.Max(0, _singleton._buffer.Length - 1));
```

**Review by rjmholt — APPROVED** [review]

**Comment by msftrncs** [comment]
@springcomp, looks like something went wrong with line endings or something, in your last commit.  I keep running in to this in certain places as well.

**Comment by springcomp** [comment]
Sorry about that, I did make the change using GitHub’s web interface but I should have been more careful.

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**